gnomad/utils/vcf.py

```diff
@@ -865,3 +890,64 @@ def set_female_y_metrics_to_na(
             }
         )
     return female_metrics_dict
+
+
+def build_vcf_export_reference(
+    name: str,
+    build: str = "GRCh38",
+    contigs_keep: List[str] = [f"chr{i}" for i in range(1, 23)]
+    + ["chrX", "chrY", "chrM"],
+) -> hl.ReferenceGenome:
+    """
+    Create export reference based on reference genome defined by `build`.
+
+    By default this will return a new reference with all non-standard contigs eliminated. Keeps chr 1-22, Y, X, and M.
+
+    Non-standard for example is: ##contig=<ID=chr3_GL000221v1_random,length=155397,assembly=GRCh38>
+
+    :param name: Name to use for new reference.
+    :param build: Reference genome build to use as starting reference genome.
+    :param contigs_keep: Contigs to keep from reference genome defined by `build`.
+    :return: Reference genome for VCF export containing only contigs in `contigs_keep`
+    """
+    ref = hl.get_reference(build)
+
+    export_reference = hl.ReferenceGenome(
+        name=name,
+        contigs=contigs_keep,
+        lengths={contig: ref.lengths[contig] for contig in contigs_keep},
+        x_contigs=ref.x_contigs,
+        y_contigs=ref.y_contigs,
+        par=[
+            (interval.start.contig, interval.start.position, interval.end.position)
+            for interval in ref.par
+        ],
+        mt_contigs=ref.mt_contigs,
```

Comment:
just curious, is this still necessary if `contigs_keep` doesn't specify chrM?

Reply:
I don't think so, but it will just return an empty list which is what the default is for initialization, so it will still work